src/BigInteger.php

```diff
@@ -222,6 +222,18 @@ public function subtract(string $value): BigInteger
     }
 
     /**
+     * Checks if the big integr is the prime number probably.
```

Comment:
Typo in `integr`, should be `integer`.

Maybe also remove the word "probably"? We could change that into a parameter:

```
/**
 * Checks if the big integer is a prime number.
 *
 * @param float $probabilityFactor A normalized factor between 0 and 1 used for checking the probability.
 * @return bool Returns true if the number is a prime number false if not.
 */
```

According to https://secure.php.net/manual/en/function.gmp-prob-prime.php that value should be between 5 and 10 so if we use a normalized value, it becomes more accessible.

```
public function isPrimeNumber(float $probabilityFactor = 1.0): bool
{
    // Always scale down since we need an integer, 0 means 5, 0.2 means 6, etc.
    $reps = floor(($probabilityFactor * 5.0) + 5.0);

    return gmp_prob_prime($this->value, $reps) !== 0;
}
```

What do you think?

Reply:
I think this approach is ok.
Do we have to output the message when the ```$reps``` is greater than 10?

Reply:
Good one! What about throwing a `PHP\Math\BigInteger\Exception\InvalidArgumentException` if the value is out of range?

Reply:
Or maybe https://secure.php.net/manual/en/class.rangeexception.php is more suitable.